com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/resources/jsweb/dmloader.js

```diff
@@ -484,21 +482,37 @@ var GameArchiveLoader = {
         ProgressUpdater.setupTotal(totalSize + EngineLoader.wasm_instantiate_progress);
     },
 
-    downloadContent: function() {
+    downloadContent: async function() {
         var file = this._files[this._fileIndex];
 
         if (Module['isDMFSSupported']) {
             const path = `${DMSYS.GetUserPersistentDataRoot()}/${file.name}`;
             try { // see if already and stored
                 const stat = FS.stat(path);
                 if(stat) {
-                    if (file.size == stat.size) { // should we verify hash too?
+                    let matches = (file.size == stat.size)
+                    if (matches && file.sha1) {
+                        const stream = FS.open(path, "r");
+                        if(stream) {
+                            try {
+                                const mmap = FS.mmap(stream, stat.size, 0, 0x01, 0x01); //PROT_READ, MAP_SHARED
+                                if(mmap) {
+                                    const digest = await window.crypto.subtle.digest("SHA-1", mmap);
+                                    matches = Array.from(new Uint8Array(digest)).map(b => b.toString(16).padStart(2, '0')).join('') == file.sha1;
+                                }
+                            } catch(e) { }
+                            FS.close(stream);
+                        } else {
+                            matches = false;
+                        }
+                    }
+                    if (matches) {
```

Comment:
What kind of impact will this verification have on load times? We do not wish to slow down loading of content if we can avoid it. Can we hide this behind an option in game.project?

Reply:
Certainly adds some cost, I've pushed change to have it be optional

Reply:
Great, but how do I disable it? We need a game.project setting and a check in HTML5Bundler too don't we?

You need an entry in com/dynamo/bob/meta.properties and in editor/resources/edn.properties. The default value should be off.

Reply:
Yes, I made the default value off. I am just passing --with-sha1 into bob.jar currently. Do you want it to show up in the editor?

Reply:
Now that I think about, it's probably fine that it is an option only available in command line builds. But the name of the option doesn't tell a user that it only applies to html5 builds. We should at least write that when printing the help.